weaveworks/weave-gitops
# implement vcluster connect capability

Fixes #3451

This pull request adds the `connect` feature to vcluster by incorporating code from several vcluster packages. The following packages were utilized:

  - `connect`: enables connecting to a virtual cluster. This package is a rewritten of `cmd` connect package.
  - `find`: provides a way to locate vcluster resources (can be reused from `gitops remove run`)
  - `kubeconfig`: manages Kubernetes configuration files
  - `localkubernetes`: provides a way to interact with a local Kubernetes cluster
  - `portforward`: enables port forwarding (can be reused later for GitOps Run)

With these packages, the connect feature allows users to easily connect to a virtual cluster and interact with its resources.

Following up tickets would be:
- reuse `find` for `gitops remove run`
- reuse `portforward` for `gitops run`

# How to test
  - Run `gitops run` in the session mode, and it should work normally just like before the removal of the vcluster packages.

# TODO
  - Write docs on how to patch these files when something is broken in the future.
